File: src/Qwack.Transport/TransportObjects/MarketData/Models/TO_AssetFxModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using ProtoBuf;
using Qwack.Transport.TransportObjects.Instruments;
using Qwack.Transport.TransportObjects.MarketData.Correlations;
using Qwack.Transport.TransportObjects.MarketData.Curves;
using Qwack.Transport.TransportObjects.MarketData.VolSurfaces;

namespace Qwack.Transport.TransportObjects.MarketData.Models
{
    [ProtoContract]
    public class TO_AssetFxModel
    {
        [ProtoMember(8)]
        public Dictionary<TO_VolSurfaceKey, TO_VolSurface> AssetVols{ get; set; }
        [ProtoMember(9)]
        public Dictionary<string, TO_PriceCurve> AssetCurves { get; set; }
        [ProtoMember(10)]
        public Dictionary<string, TO_FixingDictionary> Fixings { get; set; }
        [ProtoMember(11)]
        public DateTime BuildDate { get; set; }
        [ProtoMember(12)]
        public TO_FundingModel FundingModel { get; set; }
        [ProtoMember(13)]
        public TO_CorrelationMatrix CorrelationMatrix { get; set; }
        [ProtoMember(14)]
        public TO_Portfolio Portfolio { get; set; }

    }
}

File: src/Qwack.Transport/GeneratedTypeScript/TO_AssetFxModel.ts

import { TO_FundingModel } from './TO_FundingModel';
import { TO_CorrelationMatrix } from './TO_CorrelationMatrix';
import { TO_Portfolio } from './TO_Portfolio';
import { TO_VolSurface } from './TO_VolSurface';
import { TO_PriceCurve } from './TO_PriceCurve';
import { TO_FixingDictionary } from './TO_FixingDictionary';

export interface TO_AssetFxModel  {
	 assetVols: { [key: string]: TO_VolSurface; };
	 assetCurves: { [key: string]: TO_PriceCurve; };
	 fixings: { [key: string]: TO_FixingDictionary; };
	 buildDate: Date;
	 fundingModel: TO_FundingModel;
	 correlationMatrix: TO_CorrelationMatrix;
	 portfolio: TO_Portfolio;
}


